plugins/glustershd/actor.go

```diff
@@ -0,0 +1,93 @@
+package glustershd
+
+import (
+	"errors"
+
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/glusterd2/xlator"
+	gderrors "github.com/gluster/glusterd2/pkg/errors"
+)
+
+var names = [...]string{"replicate", "afr"}
+
+type shdActor struct{}
+
+func (actor *shdActor) Do(v *volume.Volinfo, key string, value string) error {
+	// validate volume type
+	if !isVolReplicate(v.Type) {
+		return errors.New("volume Type not supported")
+	}
+	glustershDaemon, err := newGlustershd()
+	if err != nil {
+		return err
+	}
+	switch value {
+	case "on":
+		if isHealEnabled(v, key) {
+			err = daemon.Start(glustershDaemon, true)
+			if err == gderrors.ErrProcessAlreadyRunning {
+				return nil
+			}
```

Comment:
You should check for nil too. And also return error if the error is other than `gderrors.ErrProcessAlreadyRunning`